capi/geos_ts_c.cpp

```diff
@@ -4411,24 +4411,36 @@ extern "C" {
         try {
             using geos::geom::LinearRing;
 
-            auto vholes = geos::detail::make_unique<std::vector<LinearRing*>>(nholes);
-
+            /*
+            * First check that we can cast these generic
+            * geometry pointers into linearrings. Then
+            * convert them to unique_ptr to pass to the constructor
+            * so we can get a no-copy transfer of the data
+            */
+            std::vector<LinearRing*> tmpholes(nholes);
             for (size_t i = 0; i < nholes; i++) {
-                (*vholes)[i] = dynamic_cast<LinearRing*>(holes[i]);
-                if ((*vholes)[i] == nullptr) {
+                LinearRing* lr = dynamic_cast<LinearRing*>(holes[i]);
+                if (! lr) {
                     handle->ERROR_MESSAGE("Hole is not a LinearRing");
                     return NULL;
                 }
+                tmpholes[i] = lr;
             }
-
             LinearRing* nshell = dynamic_cast<LinearRing*>(shell);
             if(! nshell) {
                 handle->ERROR_MESSAGE("Shell is not a LinearRing");
                 return NULL;
             }
             const GeometryFactory* gf = handle->geomFactory;
 
-            return gf->createPolygon(nshell, vholes.release());
+            /* Create unique_ptr version for constructor */
+            std::vector<std::unique_ptr<LinearRing>> vholes(nholes);
```

Comment:
Unless I'm wrong the current code will have vholes with nholes first elements at nullptr, followed by the content of tmpholes.
You'd probably want to do instead
```suggestion
            std::vector<std::unique_ptr<LinearRing>> vholes;
            vholes.reserve(nholes);
```

Reply:
yep, you are right